docs/design/reserve-action.md

```diff
@@ -0,0 +1,278 @@
+sorted in descending order of rank.
+
+In the process of calculating node scores based on node idle resource, there will be `cpu` `memory` and `scalar resources` 
+participating in the score calculation. The formula for calculating score for a single resource is  
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=100-\frac{100*usedResource}{allocatableResource}" style="border:none;">
+
+The formula for calculating the node score based on the above three resources is
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=\frac{\frac{cpuScore%2b memoryScore}{2}%2b2*sclarScore}{3}" style="border:none;">
+
+In the process of `reduce`, the node score will be re-adjusted to ten points or zero points according to the threshold of 
+the number of reserved nodes.
+
+### Reserve logic
+
+Even if the `reserve` action is added, it it still necessary to ensure that the jobs with higher priority have priority to 
+be scheduled and get resources in the cluster. When attempting to `reserve` resources for a starving job, it is necessary 
+to ensure that in the case of resource situation of current cluster, any job in the job queue who has a higher priority 
+than the starving job cannot be scheduled. If a job with a higher priority than the starving job can be scheduled successfully
+in this round, the starving job need to wait until the high-priority job is scheduled. It need to wait for the next schedule
+round to try to reserve resources on nodes. 
+
+![reserve-action-logic](../../docs/images/reserve-action-logic.png)
+
+## Data Model
+
+Introduce one node resource `Reserved` in `NodeInfo`, used to indicate the size of resources reserved on the node. The size 
+of the reserved resource on the node is the sum of the resource requests of all the pods reserved on the node. The 
+reserved resources should be smaller than the allocatable resources.
+```go
+	
+// NodeInfo is node level aggregated information.
+type NodeInfo struct {
+	...
+
+	// Reserved is reserved resources for starving task
+	Reserved *Resource
+
+	...
+}
+
+```
+
+Introduce two task status `ForceAllocated` and `FakeAllocated`. The `ForceAllocated` indicates that the pod is assigned 
+to this node, occupies the corresponding resources of the node, and does not allow other tasks to backFill these resources.
+The `FakeAllocated` indicates that  although the task is added to the node, it does not actually occupy the resources of 
+the node, it just wants the node to reserve resources for it. 
+
+```go
+// FakeAllocated means a task/pod is fake allocated to the node
+// this is the task is added to the node, but not allocated resources for it
+FakeAllocated
+
+// ForceAllocated means a task/pod is allocated to the node
+ForceAllocated
+```
+The `Reserved` resources on the node will be increased to 
+indicate that the node reserves resources for the tasks, at the same time, the idle resources of the node will be reduced 
```

Comment:
> the idle resources of the node will be reduced to prevent the node from providing resources for other pods.

IMO, that maybe not necessary to modify idle resources.

Reply:
Updating the `idle` resources of node is to avoid subsequent actions to dispatch tasks to this node. However, in order to ensure that the logic of node score calculation based on the idle resource is fair to all the tasks, it is necessary to ensure that the idle resources of the node are stable during the `reserve` process. I will try a new logic to ensure that the node who has reserve resources no longer dispatches other tasks by other logic, instead of operating the `idle` resources of the node.

Reply:
The idle resource of node is no longer updated for the `reserved` task. Using the `Reserved` resource of the node to indicate whether the node can schedule a new non-BestEffort task.